gateway-js/src/index.ts

```diff
@@ -238,10 +239,23 @@ export class ApolloGateway implements GraphQLService {
     // 3. If config is `undefined`, use the default uplink URL
     if (isManagedConfig(this.config)) {
       const envEndpoint = process.env.APOLLO_SCHEMA_CONFIG_DELIVERY_ENDPOINT;
-      this.schemaConfigDeliveryEndpoint =
-        this.config.schemaConfigDeliveryEndpoint ??
-        envEndpoint ??
-        'https://uplink.api.apollographql.com/';
+      const endPoints = envEndpoint?.split(",") ?? null;
+      if (this.config.schemaConfigDeliveryEndpoint && !this.config.uplinkEndpoints) {
+        this.config.uplinkEndpoints = [this.config.schemaConfigDeliveryEndpoint];
+      }
+
+      this.schemaConfigDeliveryEndpoints =
+        this.config.uplinkEndpoints ??
+          endPoints ?? [
+            'https://uplink.api.apollographql.com/',
+            'https://aws.uplink.api.apollographql.com/'
+          ];
+
+      if (this.config.uplinkMaxRetries != null) {
+        this.schemaDeliveryMaxRetries = this.config.uplinkMaxRetries;
+      } else {
+        this.schemaDeliveryMaxRetries = this.schemaConfigDeliveryEndpoints.length * 3;
+      }
```

Comment:
Can simplify this a bit with nullish coalesce (logically equivalent to the `!= null` check you're performing)
```suggestion
      this.uplinkDeliveryMaxRetries = this.config.uplinkMaxRetries ?? this.schemaConfigDeliveryEndpoints.length * 3;
```